src/core/Icon/Icon.tsx

```diff
@@ -60,20 +44,7 @@ const StyledSuomifiIcon = styled(
   ${iconBaseStyles}
 `;
 
-/**
- * General icon-component
- */
-export class Icon extends Component<IconProps> {
-  render() {
-    const { icon, ...passProps } = this.props;
-    const { className, ariaLabel } = this.props;
-
-    if (icon !== undefined) {
-      return <StyledSuomifiIcon {...passProps} icon={icon} />;
-    }
-
-    iconLogger(ariaLabel, className);
```

Comment:
For the `<Icon>` component, I simplified the outer layer a lot. Since the `icon` prop is mandatory, it really makes no sense to have this if-condition there and have it return undefined if it's not provided. Same goes for `<StaticIcon>` and `<LogoIcon>`.

BTW, tested this without TS and if the icon prop is left out it renders as the login-icon